core/src/com/unciv/UncivGame.kt

```diff
@@ -172,6 +175,106 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
+                        else -> "Could not upload game!"
+                    }
+                    launchOnGLThread { // Since we're changing the UI, that should be done on the main thread
+                        val cantUploadNewGamePopup = Popup(worldScreen!!)
+                        cantUploadNewGamePopup.addGoodSizedLabel(message).row()
+                        cantUploadNewGamePopup.addCloseButton()
+                        cantUploadNewGamePopup.open()
+                    }
+                    isTurnProcessing = false
+                    worldScreen?.shouldUpdate = true
+                    return@runOnNonDaemonThreadPool
+                }
+            }
+
+            isTurnProcessing = false
+
+            if (gameInfo != originalGameInfo)
+                return@runOnNonDaemonThreadPool  // while this was turning we loaded another game
+
+            debug("Next turn took %sms", System.currentTimeMillis() - startTime)
+            reloadWorldScreen(gameInfoClone)
+        }
+    }
+
+    /** This exists so that no reference to the current world screen remains, so the old world screen can get garbage collected during [loadGame]. */
+    fun reloadWorldScreen(
+        newGameInfo: GameInfo? = null,
+        autoSaveDisabled: Boolean = false,
+        replaceOnly: Boolean = false) {
+
+        val gameInfoToLoad = when {
+            replaceOnly && gameInfo != null && worldScreen != null -> gameInfo
+            !replaceOnly && newGameInfo != null -> newGameInfo
+            else -> null
+        } ?: return
+
+        Concurrency.run {
+            try {
+
+                val newWorldScreen = loadGame(gameInfoToLoad)
+                val shouldAutoSave = !autoSaveDisabled
+                        && gameInfoToLoad.turns % settings.turnsBetweenAutosaves == 0
+                if (shouldAutoSave)
+                    newWorldScreen.autoSave()
+
+            } catch (notAPlayer: UncivShowableException) {
+
+                withGLContext {
+                    val (message) = LoadGameScreen.getLoadExceptionMessage(notAPlayer)
+                    val mainMenu = goToMainMenu()
+                    ToastPopup(message, mainMenu)
+                }
+                return@run
+
+            } catch (oom: OutOfMemoryError) {
+
+                withGLContext {
+                    val mainMenu = goToMainMenu()
+                    ToastPopup("Not enough memory on phone to load game!", mainMenu)
+                }
+                return@run
+
+            }
+        }
+    }
+
+    /** Reload this Popup after major changes (resolution, tileset, language, font) */
+    fun reloadWorldAndOptions(page: Int = 0) {
+        settings.save()
+        BaseScreen.setSkin()
+
+        val screen = getScreen()
+
+        if (screen is WorldScreen)
+            reloadWorldScreen(replaceOnly = true)
+        else if (screen is MainMenuScreen)
+            replaceCurrentScreen(MainMenuScreen())
+
+        screen?.openOptionsPopup(page)
+    }
```

Comment:
Sounds like a function for the Options menu, this is extremely specialized, not for UncivGame for sure

Reply:
Options are not responsible for WorldScreen's or any screen other than itself. Any logic that involves recreating worldscreen belong to a parent which hold WorldScreen, that is to UncivGame

Reply:
In this case I agree, since Options is a cross-screen and game-wide functionality